File: mirea/contests/2021-09-26 MIREA Welcome Contest 2021/NoPoint C++/Задача F.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;
#define int ll

int32_t main() {
  int tests;
  cin >> tests;
  for(int test = 0; test < tests; ++test) {
    int n, k, d;
    cin >> n >> k >> d;
    vector<int> arr(n);
    for(int i = 0; i < n; ++i) {
      cin >> arr[i];
    }

    vector<vector<int>> costs(d);
    for(int i = 0; i < n; ++i) {
      costs[i % d].push_back(arr[i]);
    }
    int ans = 0;
    for(int i = 0; i < d; ++i) {
      vector<int> ps(costs[i].size() + 1);
      partial_sum(costs[i].begin(), costs[i].end(), next(ps.begin()));
      auto get_sum = [&ps](int left, int right) {
        return ps[right] - ps[left];
      };
      for(int j = k; j <= (int)costs[i].size(); ++j) {
        ans = max(ans, get_sum(j - k, j));
      }
    }

    cout << ans << endl;
  }
}

File: mirea/contests/2021-09-26 MIREA Welcome Contest 2021/NoPoint Python/Задача F.py
def main():
  tests = int(input())
  for test in range(tests):
    n, k, d = map(int, input().split())
    arr = [int(num) for num in input().split()]

    costs = [[] for _ in range(d)]
    ans = 0
    for i, val in enumerate(arr):
      costs[i % d].append(val)

    for i in range(d):
      elems = len(costs[i])
      ps = [0] * (elems + 1)
      for j in range(elems):
        ps[j + 1] = ps[j] + costs[i][j]

      def get_sum(left, right):
        return ps[right] - ps[left]

      for j in range(k, elems + 1):
        ans = max(ans, get_sum(j - k, j))

    print(ans)


main()

